crates/sparrow-execution/src/source_tasks.rs

```diff
@@ -0,0 +1,65 @@
+use error_stack::ResultExt;
+use futures::stream::BoxStream;
+use futures::StreamExt;
+use sparrow_arrow::Batch;
+use sparrow_scheduler::{Injector, PipelineInputs};
+
+use crate::Error;
+
+/// Manages the async streams that must be read for input.
+#[derive(Default)]
+pub(super) struct SourceTasks {
```

Comment:
Added this to collect the source streams and pipeline inputs for the purposes of running them in tokio.